lib/ansible/utils/collection_loader/_collection_finder.py

```diff
@@ -78,6 +84,101 @@ def is_python_identifier(self):  # type: (str) -> bool
 
 
 PB_EXTENSIONS = ('.yml', '.yaml')
+SYNTHETIC_PACKAGE_NAME = '<ansible_synthetic_collection_package>'
+
+
+class _AnsibleNSTraversable:
+    def __init__(self, *paths):
+        self._paths = [pathlib.Path(p) for p in paths]
+
+    def __repr__(self):
+        return '_AnsibleNSTraversable(\'%s\')' % '\', \''.join(map(to_text, self._paths))
+
+    def iterdir(self):
+        return itertools.chain(*[list(p.iterdir()) for p in self._paths if p.is_dir()])
```

Comment:
Anything preventing us from using a nicer interface to this?
```suggestion
        return itertools.chain.from_iterable(p.iterdir() for p in self._paths if p.is_dir())
```